run-make-check.sh

```diff
@@ -76,28 +76,26 @@ function run() {
 
     CMAKE_BUILD_OPTS="-DWITH_FIO=ON -DWITH_GTEST_PARALLEL=ON"
 
-    # Are we in the CI ?
+    # The following settings are made for improving ccache efficiency
+    # by removing the entropy generated by the date/time embedded in the build
+    CMAKE_BUILD_OPTS="$CMAKE_BUILD_OPTS -D ENABLE_GIT_VERSION=OFF"
+    export SOURCE_DATE_EPOCH=$(date +%D |date -f- +%s)
```

Comment:
moving SOURCE_DATE_EPOCH and disabling the GIT versioning out of the JENKINS condition could be dangerous if people are not aware of it as the resulting binaries have a broken tracability.

Reply:
The purpose of `run-make-check.sh` is to make it easy for folks to run "make check" and, in my mind, that includes making it easy for them to run it as fast as possible. If these ccache tweaks bring an order-of-magnitude speed-up, that's something users need to have by default.

Reply:
Why not but in that case we have to warn them explicitly that the generated binaries doesn't have the proper tractability for a production use. Printing a visible message would be fine for me.

Reply:
Visible message added - please take a look.